op-service/sources/l1_client.go

```diff
@@ -255,6 +257,76 @@ func (s *L1Client) ClearReceiptsCacheBefore(blockNumber uint64) {
 	s.recProvider.GetReceiptsCache().RemoveLessThan(blockNumber)
 }
 
+func (s *L1Client) GetBlobs(ctx context.Context, ref eth.L1BlockRef, hashes []eth.IndexedBlobHash) ([]*eth.Blob, error) {
+	if len(hashes) == 0 {
+		return []*eth.Blob{}, nil
+	}
+
+	blobSidecars, err := s.getBlobSidecars(ctx, ref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get blob sidecars for L1BlockRef %s: %w", ref, err)
+	}
+
+	validatedBlobs, err := validateBlobSidecars(blobSidecars, ref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate blob sidecars for L1BlockRef %s: %w", ref, err)
+	}
+
+	blobs := make([]*eth.Blob, len(hashes))
+	for i, indexedBlobHash := range hashes {
+		blob, ok := validatedBlobs[indexedBlobHash.Hash]
+		if !ok {
+			return nil, fmt.Errorf("blob sidecars fetched from rpc mismatched with expected hash %s for L1BlockRef %s", indexedBlobHash.Hash, ref)
+		}
+		blobs[i] = blob
+	}
+	return blobs, nil
+}
+
+func (s *L1Client) getBlobSidecars(ctx context.Context, ref eth.L1BlockRef) (eth.BSCBlobSidecars, error) {
+	var blobSidecars eth.BSCBlobSidecars
+	err := s.client.CallContext(ctx, &blobSidecars, "eth_getBlobSidecars", numberID(ref.Number).Arg())
+	if err != nil {
```

Comment:
We need to determine: When blobSidecar cannot be found, the behavior of the BSC interface is to not return an error, but instead return blobSidecars == nil. If it is the opposite, returning an error, we need to ensure that it returns ethereum.NotFound as the error. If ethereum.NotFound cannot be returned at the appropriate time, it may cause the blob parsing process to get stuck in an infinite loop.